irods/test/data_obj_test.py

```diff
@@ -2044,6 +2044,26 @@ def test_append_mode_will_append_to_data_object__issue_495(self):
             if data.exists(testfile):
                 data.unlink(testfile,force=True)
 
+    def test_client_redirect_lets_go_of_connections__issue_562(self):
+        self._skip_unless_connected_to_local_computer_by_other_than_localhost_synonym()
+        # Force data object connections to redirect by enforcing a non-equivalent hostname for their resource
+        with self.create_simple_resc(hostname = 'localhost') as resc_name:
+            REPS_TO_REPRODUCE_CONNECT_ERROR = 100
+            paths=[]
+            try:
+                # Try to exhaust connections
+                for n in range(REPS_TO_REPRODUCE_CONNECT_ERROR):
+                    data_path = '{self.coll_path}/issue_562_test_obj_{n:03d}.dat'.format(**locals())
+                    paths.append(data_path)
+                    with self.sess.data_objects.open(data_path, 'w', **{kw.DEST_RESC_NAME_KW: resc_name}) as f:
+                        pass
```

Comment:
is the lack of exploding is good enough?

do we need an assertion or two in here?

Reply:
Maybe. Could just check the number of connections in self.sess.active is not climbing.

Reply:
yes - can discuss more.   perhaps not exploding is enough - can you assert here in the comments that it ALWAYS exploded with this test before the fix?

Reply:
Yes, will do that

Reply:
> Maybe. Could just check the number of connections in self.sess.active is not climbing.

This sounds like a good assertion to make during and after all the opens... Even if it doesn't explode, we could see that it was "leaking" connections, right?

And yes, please confirm alwaysexplodes.

Reply:
> And yes, please confirm always explodes.

@alanking , you meant confirm by adding a comment affirming that it explodes, correct? Or did you mean re-write the test so it is always guaranteed to trip the exception?

Reply:
Oh, I suppose I originally meant that I wanted to confirm that it always exploded before the fix, but adding a code comment affirming as such would be good, yes. I trust your assertion of alwaysexplodes so just adding a comment should be fine.

Reply:
I can confirm at least verbally, here and now, that in all my past experiments it did in fact explode.

Reply:
Added the comment to reaffirm an "explosion" problem.